plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/AbstractTestTrinoS3FileSystemObjectStorage.java

```diff
@@ -0,0 +1,546 @@
+                assertThat(paths).containsOnly(directoryKey + PATH_SEPARATOR);
+
+                assertTrue(fs.delete(new Path(prefixPath), true));
+
+                assertThat(listPaths(fs.getS3Client(), getBucketName(), prefix, true)).isEmpty();
+            }
+            finally {
+                fs.delete(new Path(prefixPath), true);
+            }
+        }
+    }
+
+    @Test
+    public void testDeleteRecursivelyPrefixContainingOneDirectoryWithOneObject()
+            throws Exception
+    {
+        String prefix = "test-delete-recursively-path-one-directory-with-one-object-" + randomUUID();
+        String prefixPath = "s3://%s/%s".formatted(getBucketName(), prefix);
+
+        try (TrinoS3FileSystem fs = new TrinoS3FileSystem()) {
+            fs.initialize(new URI("s3://%s/".formatted(getBucketName())), defaultS3Configuration());
+
+            try {
+                String tableKey = prefix + "/table";
+                String partitionKey = tableKey + "/partition=1";
+                createDirectory(fs.getS3Client(), getBucketName(), partitionKey);
+                String filename = "file.txt";
+                fs.createNewFile(new Path("s3://%s/%s".formatted(getBucketName(), partitionKey), filename));
+
+                assertTrue(fs.delete(new Path(prefixPath), true));
+
+                assertThat(listPaths(fs.getS3Client(), getBucketName(), prefix, true)).isEmpty();
+            }
+            finally {
+                fs.delete(new Path(prefixPath), true);
+            }
+        }
+    }
+
+    @Test
+    public void testDeleteRecursivelyPrefixContainingMultipleObjectsPlain()
+            throws Exception
+    {
+        String prefix = "test-delete-recursively-path-multiple-objects-plain-" + randomUUID();
+        String prefixPath = "s3://%s/%s".formatted(getBucketName(), prefix);
+
+        try (TrinoS3FileSystem fs = new TrinoS3FileSystem()) {
+            fs.initialize(new URI("s3://%s/".formatted(getBucketName())), defaultS3Configuration());
+
+            try {
+                String filename1 = "file1.txt";
+                String filename2 = "file2.txt";
+                fs.createNewFile(new Path(prefixPath, filename1));
+                fs.createNewFile(new Path(prefixPath, filename2));
+                List<String> paths = listPaths(fs.getS3Client(), getBucketName(), prefix, true);
+                assertThat(paths).containsOnly(
+                        "%s/%s".formatted(prefix, filename1),
+                        "%s/%s".formatted(prefix, filename2));
+
+                assertTrue(fs.delete(new Path(prefixPath), true));
+
+                assertThat(listPaths(fs.getS3Client(), getBucketName(), prefix, true)).isEmpty();
+            }
+            finally {
+                fs.delete(new Path(prefixPath), true);
+            }
+        }
+    }
+
+    @Test
+    public void testDeleteRecursivelyPrefixCorrespondingToDirectoryObjectContainingMultipleObjectsSpread()
```

Comment:
as in other comment -- try to make test method names a bit shorter, while still meaningful